### PR TITLE
Implement RAII for `PreservationTable`

### PR DESCRIPTION
**Pull request by JosiahWI**
This is pretty straightforward, but the catch is what to do if there are unreleased evacuation blocks at destruction. I still need to write code to handle printing a warning or something to that effect before this is fully ready.

### CODE REVIEW

File: src/iocore/cache/StripeSM.cc [hunk 1]
```diff
@@ -109,13 +109,9 @@ struct StripeInitInfo {
   }
 };
 
-StripeSM::StripeSM(CacheDisk *disk, off_t blocks, off_t dir_skip) : Continuation(new_ProxyMutex()), Stripe{disk, blocks, dir_skip}
+StripeSM::StripeSM(CacheDisk *disk, off_t blocks, off_t dir_skip)
+  : Continuation(new_ProxyMutex()), Stripe{disk, blocks, dir_skip}, _preserved_dirs{static_cast<int>(len)}
```
Comment (JosiahWI) [inline src/iocore/cache/StripeSM.cc:113]: What's the right way to do this in C++? This makes me nervous because it depends on initialization order.

**Comment by JosiahWI** [comment]
The AuTest bad chunked encoding failed.

```
     file /tmp/sandbox/bad_chunked_encoding/_output/bad_chunked_encoding-verifier-server2/stream.stdout.txt : Verify that the body never got through. - Failed
        Reason: Contents of /tmp/sandbox/bad_chunked_encoding/_output/bad_chunked_encoding-verifier-server2/stream.stdout.txt contains expression: "abc"
           Details:
             Via: http/1.1 traffic_server[ed350727-0c1d-4e42-9a5d-aeca5abc9853] (ApacheTrafficServer/10.1.0) : 17
             Via: http/1.1 traffic_server[ed350727-0c1d-4e42-9a5d-aeca5abc9853] (ApacheTrafficServer/10.1.0) : 35
             Via: http/1.1 traffic_server[ed350727-0c1d-4e42-9a5d-aeca5abc9853] (ApacheTrafficServer/10.1.0) : 53
             Via: http/1.1 traffic_server[ed350727-0c1d-4e42-9a5d-aeca5abc9853] (ApacheTrafficServer/10.1.0) : 71
             Via: http/1.1 traffic_server[ed350727-0c1d-4e42-9a5d-aeca5abc9853] (ApacheTrafficServer/10.1.0) : 95
             Via: http/1.1 traffic_server[ed350727-0c1d-4e42-9a5d-aeca5abc9853] (ApacheTrafficServer/10.1.0) : 119
  ```

**Comment by JosiahWI** [comment]
Not rerunning CI since this is a draft and I already got enough confidence that AuTest found no problems with it.

File: src/iocore/cache/StripeSM.cc (hunk 1 above)
Comment (JosiahWI) [reply src/iocore/cache/StripeSM.cc:116]: I've added a comment explaining the weirdness.

**Comment by JosiahWI** [comment]
[approve ci rat]

**Comment by JosiahWI** [comment]
[approve ci ubuntu rocky osx freebsd fedora debian clang-analyzer centos autest]

**Comment by JosiahWI** [comment]
The last test run of the AuTest remap_acl failed.

**Comment by JosiahWI** [comment]
[approve ci autest]

**Review by moonchen — APPROVED** [review]
Nice.

**Pull request merged**